core/src/commonJava/kotlin/com/powersync/db/JdbcSqliteDriver.kt

```diff
@@ -118,34 +93,49 @@ internal class PSJdbcSqliteDriver(
             stmt.executeQuery(mapper)
         }
 
-    internal fun loadExtensions(vararg extensions: Pair<Path, String>) {
+    internal fun loadExtensions(vararg extensions: Pair<String, String>) {
         connection.database.enable_load_extension(true)
         extensions.forEach { (path, entryPoint) ->
             val executed =
                 connection.prepareStatement("SELECT load_extension(?, ?);").use { statement ->
-                    statement.setString(1, path.absolutePathString())
+                    statement.setString(1, path)
                     statement.setString(2, entryPoint)
                     statement.execute()
                 }
-            check(executed) { "load_extension(\"${path.absolutePathString()}\", \"${entryPoint}\") failed" }
+            check(executed) { "load_extension(\"${path}\", \"${entryPoint}\") failed" }
         }
         connection.database.enable_load_extension(false)
     }
 
-    internal fun enableWriteAheadLogging() {
-        val executed = connection.prepareStatement("PRAGMA journal_mode=WAL;").execute()
-        check(executed) { "journal_mode=WAL failed" }
+    private inner class Transaction(
+        override val enclosingTransaction: Transaction?,
+    ) : Transacter.Transaction() {
+        init {
+            connection.prepareStatement("BEGIN TRANSACTION").use(PreparedStatement::execute)
```

Comment:
This doesn't support nested transactions. I think we should either:

1. Keep some kind of depth counter on this `Transaction` class to emit `SAVEPOINT` / `RELEASE` / `ROLLBACK TO` statements on inner transactions.
2. Assert that `enclosingTransaction` is null, because ending this inner transaction would otherwise and the enclosing one?

Reply:
We currently set `noEnclosing = true` whenever we start a transaction, but I have also now added an assertion in the driver.